controller/api/public/grpc_server.go

```diff
@@ -176,31 +189,104 @@ func (_ *grpcServer) Version(ctx context.Context, req *pb.Empty) (*pb.VersionInf
 }
 
 func (s *grpcServer) ListPods(ctx context.Context, req *pb.Empty) (*pb.ListPodsResponse, error) {
-	resp, err := s.telemetryClient.ListPods(ctx, &telemPb.ListPodsRequest{})
+	log.Debugf("ListPods request: %+v", req)
+
+	// Reports is a map from instance name to the absolute time of the most recent
+	// report from that instance.
+	reports := make(map[string]time.Time)
+
+	// Query Prometheus for all pods present
+	vec, err := s.queryProm(ctx, podQuery)
+	if err != nil {
+		return nil, err
+	}
+	for _, sample := range vec {
+		pod := string(sample.Metric["pod"])
+		timestamp := sample.Timestamp
+
+		reports[pod] = time.Unix(0, int64(timestamp)*int64(time.Millisecond))
+	}
+
+	pods, err := s.podLister.List(labels.Everything())
 	if err != nil {
 		return nil, err
 	}
-	return resp, nil
+	podList := make([]*pb.Pod, 0)
+
+	for _, pod := range pods {
+		if s.shouldIgnore(pod) {
+			continue
+		}
+
+		deployment, err := s.getPodOwner(pod)
+		if err != nil {
+			log.Debugf("Cannot get deployment for pod %s: %s", pod.Name, err)
+			deployment = ""
+		}
+
+		updated, added := reports[pod.Name]
+
+		status := string(pod.Status.Phase)
+		if pod.DeletionTimestamp != nil {
+			status = "Terminating"
+		}
+
+		controllerComponent := pod.Labels[pkgK8s.ControllerComponentLabel]
+		controllerNS := pod.Labels[pkgK8s.ControllerNSLabel]
+
+		item := &pb.Pod{
+			Name:                pod.Namespace + "/" + pod.Name,
+			Deployment:          deployment, // TODO: this is of the form `deployment/pod`, it should just be `deployment`
```

Comment:
As far as I can tell this field is in the form `namespace/deployment`. Maybe the comment is inaccurate?